sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/CombineFns.java

```diff
@@ -272,6 +301,60 @@ private ComposedCombineFn(
               .addAll(extractInputFns)
               .add(extractInputFn)
               .build(),
+          ImmutableList.<Optional<Coder>>builder()
+              .addAll(combineInputCoders)
+              .add(Optional.absent())
+              .build(),
+          ImmutableList.<CombineFnWithContext<?, ?, ?>>builder()
+              .addAll(fnsWithContext)
+              .add(combineFn)
+              .build(),
+          ImmutableList.<TupleTag<?>>builder().addAll(outputTags).add(outputTag).build());
+    }
+
+    /** Returns a {@link ComposedCombineFn} with an additional {@link CombineFn}. */
+    public <InputT, OutputT> ComposedCombineFn<DataT> with(
+        SimpleFunction<DataT, InputT> extractInputFn,
+        Coder combineInputCoder,
+        CombineFn<InputT, ?, OutputT> combineFn,
+        TupleTag<OutputT> outputTag) {
+      checkUniqueness(outputTags, outputTag);
+      return new ComposedCombineFn<>(
+          ImmutableList.<SerializableFunction<DataT, ?>>builder()
+              .addAll(extractInputFns)
+              .add(extractInputFn)
+              .build(),
+          ImmutableList.<Optional<Coder>>builder()
+              .addAll(combineInputCoders)
+              .add(Optional.of(combineInputCoder))
+              .build(),
+          ImmutableList.<CombineFn<?, ?, ?>>builder().addAll(combineFns).add(combineFn).build(),
+          ImmutableList.<TupleTag<?>>builder().addAll(outputTags).add(outputTag).build());
+    }
+
+    /**
+     * Returns a {@link ComposedCombineFnWithContext} with an additional {@link
+     * CombineFnWithContext}.
+     */
+    public <InputT, OutputT> ComposedCombineFnWithContext<DataT> with(
+        SimpleFunction<DataT, InputT> extractInputFn,
+        Coder combineInputCoder,
+        CombineFnWithContext<InputT, ?, OutputT> combineFn,
+        TupleTag<OutputT> outputTag) {
+      checkUniqueness(outputTags, outputTag);
+      List<CombineFnWithContext<Object, Object, Object>> fnsWithContext = Lists.newArrayList();
+      for (CombineFn<Object, Object, Object> fn : combineFns) {
+        fnsWithContext.add(CombineFnUtil.toFnWithContext(fn));
```

Comment:
nit: can this be rewritten as `combineFns.stream().map(CombilneFnUtil::toFnWithContext).collect(toList())`?